web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,426 @@
+      var stg_trg_columns = list_cols(STG_TRG_SCHEMA,STG_TRG_TBL);
+      var stg_src_columns = list_cols(STG_SRC_SCHEMA,STG_SRC_TBL);
+      if (cols_check(stg_trg_columns, stg_src_columns) === false) {
+          throw Error("ERROR: column mismatch:" + STG_TRG_TBL + "," + STG_SRC_TBL);
+      }
+
+      var stg_trg_ref = `` + STG_TRG_SCHEMA + `.` + STG_TRG_TBL;
+      var stg_src_ref = `` + STG_SRC_SCHEMA + `.` + STG_SRC_TBL;
+      var stg_del_condition = `` + STG_JOIN_KEY + ` IN
+                              (SELECT ` + STG_JOIN_KEY + ` FROM ` + stg_src_ref + `)`;
+
+      var stg_delete_stmt = `
+          DELETE FROM ` + stg_trg_ref + `
+          WHERE ` + stg_del_condition + `;`;
+
+      var stg_insert_stmt = `
+          INSERT INTO ` + stg_trg_ref + `
+            SELECT ` + stg_trg_columns + `
+            FROM ` + stg_src_ref + `;`;
+  }
+
+  // BEGIN TRANSACTION
+  snowflake.createStatement({sqlText: `BEGIN;`}).execute();
+  try {
+
+      if (SKIP_DRV !== 'TRUE') {
+          snowflake.createStatement({sqlText: drv_delete_stmt}).execute();
+          snowflake.createStatement({sqlText: drv_insert_stmt}).execute();
+      }
+      if (STAGE_NXT === 'TRUE') {
+          snowflake.createStatement({sqlText: stg_delete_stmt}).execute();
+          snowflake.createStatement({sqlText: stg_insert_stmt}).execute();
+      }
+      snowflake.createStatement({sqlText: meta_stmt}).execute();
+
+      snowflake.createStatement({sqlText: `COMMIT;`}).execute();
+
+  } catch(ERROR) {
+
+      snowflake.createStatement({sqlText: `ROLLBACK;`}).execute();
+      throw ERROR;
+
+  }
+  return "ok. commit_with_metadata succeeded.";
+
+  // == Helpers ==
+
+  function list_cols(sch,tbl) {
+      var stmt = `
+          SELECT listagg(isc.column_name, ',') WITHIN GROUP (order by isc.ordinal_position)
+          FROM information_schema.columns AS isc
+          WHERE table_schema='` + sch + `'
+            AND table_name='` + tbl + `';`;
+
+      var res = snowflake.createStatement({sqlText: stmt}).execute();
+      res.next();
+      result = res.getColumnValue(1);
+
+      return result;
+  }
+
+  function cols_check(colsStr1, colsStr2) {
+      colsArr1 = colsStr1.split(',');
+      colsArr2 = colsStr2.split(',');
+
+      // since we know there are no duplicate cols in a table,just set comparison
+      return colsArr1.every(function(c) {return colsArr2.includes(c);}) &&
+             colsArr1.length === colsArr2.length;
+  }
+
+  $$
+;
+
+
+/*
+  CTAS_SIMILAR
+  This procedure creates the target table as "similar" to the source table.
+   - Imitates BigQuery's EXCEPT(col1,...) functionality.
+   - Accepts a string of comma separated column tranformations to add
+   - and another with their respective aliases.
+
+  Input:
+  SRC_SCHEMA:    the schema of the source table
+  SRC_TBL:       the source table name
+  XCLUDE_COLS:   a string of comma-separated columns to exclude
+  TRANS_COLS:    a string of comma-separated column tranformations
```

Comment:
I think it's preferable to have the transformations be coupled with the aliases if possible, if this argument would take a list of tuples, for example.

Reply:
This was done this way in order to avoid having to map sql types to javascript ones, with varchar being the simplest. But i can certainly try.